Rubberduck.CodeAnalysis/Inspections/Extensions/IgnoreRelatedExtensions.cs

```diff
@@ -30,6 +33,21 @@ public static bool IsIgnoringInspectionResultFor(this QualifiedContext parserCon
             return parserContext.ModuleName.IsIgnoringInspectionResultFor(parserContext.Context.Start.Line, declarationFinder, inspectionName);
         }
 
+        public static bool IsIgnoringInspectionResult(this IInspectionResult result, DeclarationFinder declarationFinder)
+        {
+            switch (result)
+            {
+                case DeclarationInspectionResult dr:
+                    return dr.Target.IsIgnoringInspectionResultFor(dr.Inspection.AnnotationName);
+                case IdentifierReferenceInspectionResult irr:
+                    return irr.QualifiedName.IsIgnoringInspectionResultFor(irr.Context.Start.Line, declarationFinder, irr.Inspection.AnnotationName);
```

Comment:
This does not look right: Shouldn't it evaluate whether the reference ignores the inspection via `reference.IsIgnoringInspectionResultFor`?

To do this from the result, it might be necessary to expose the reference on this result type.

Reply:
I didn't want to extend the interface there, so I fell back to the parser context based handling, assuming that it would be functionally equivalent to the existing IdentifierReference handling.

I will adapt the IdentifierReferenceResult to expose the IdentifierReference itself. Exposing it ino the IInspectionResult interface seems inappropriate to me and we have the specific type available here.